### ISSUE TITLE
[BUG] A11y: W3C validation failures

### ISSUE THREAD

**Issue by mbmjertan** [infinum/eightshift-frontend-libs#491]
## Describe the bug

A WCAG *Level A* requirement is for all markup to be valid:
> In content implemented using markup languages, elements have complete start and end tags, elements are nested according to their specifications, elements do not contain duplicate attributes, and any IDs are unique, except where the specifications allow these features.

However, the default boilerplate setup has quite a lot of validation errors:
- [x] ID's aren't necessarily unique - we can't really do anything about this, so I'm marking it as done
- [x] ID's are sometimes empty, as in `id=""`: "Bad value for attribute id on element a: An ID must not be the empty string."
- [x] `<meta charset="" />` is output if a charset isn't set in the manifest (we should probably provide a default for that)
- [x] "A document must not include both a meta element with an http-equiv attribute whose value is content-type, and a meta element with a charset attribute."
- [x] If the `headFavicon` attribute isn't set, we output `<link rel="apple-touch-startup-image" href="">`, which is an invalid value
- [x] `outputCssVariables` has its own share of problems, such as the `style` tag not being allowed as a child element of `h2` or `div` in certain contexts. - https://github.com/infinum/eightshift-libs/pull/263
- [x] The `accordion` component uses the `aria-expanded` attribute, but doesn't have an ARIA role: "The ARIA attribute 'aria-expanded' is not valid for the element <div> with ARIA role 'none'"
- [x] This DOM node: `div.page-overlay.js-page-overlay` triggers this warning in Firefox: "Error: Clickable elements must be focusable and should have interactive semantics. [Learn more](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Keyboard#clickable_elements_must_be_focusable_and_should_have_interactive_semantics)"

This could be fixed with improvements to the default theme.

EDIT:
These issues will be fixed in Eightshift Libs:
- [x] "The type attribute is unnecessary for JavaScript resources."
- [x] "The type attribute for the style element is not needed and should be omitted."
- [x] "Element head is missing a required instance of child element title."

EDIT 2:

Remains to fix:
- [x] Bad value for attribute href on element [link](https://html.spec.whatwg.org/multipage/#the-link-element): Must be non-empty.
- [x] Attribute label not allowed on element [button](https://html.spec.whatwg.org/multipage/#the-button-element) at this point.
- [x] Attribute alt not allowed on element [picture](https://html.spec.whatwg.org/multipage/#the-picture-element) at this point.
- [x] Attribute alt not allowed on element svg at this point.
- [x] Element [div](https://html.spec.whatwg.org/multipage/#the-div-element) not allowed as child of element [button](https://html.spec.whatwg.org/multipage/#the-button-element) in this context. (Suppressing further errors from this subtree.) (3)
- [x]  The navigation role is unnecessary for element nav.
